server/src/main/java/org/opensearch/wlm/ChangeableQueryGroup.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Class to hold the fields that can be updated in a QueryGroup.
+ */
+@ExperimentalApi
+public class ChangeableQueryGroup extends AbstractDiffable<ChangeableQueryGroup> {
+
+    public static final String RESILIENCY_MODE_STRING = "resiliency_mode";
+    public static final String RESOURCE_LIMITS_STRING = "resource_limits";
+    private ResiliencyMode resiliencyMode;
+    private Map<ResourceType, Double> resourceLimits;
+
+    public static final List<String> acceptedFieldNames = List.of(RESILIENCY_MODE_STRING, RESOURCE_LIMITS_STRING);
+    private final Map<String, Function<XContentParser, Void>> fromXContentMap = Map.of(RESILIENCY_MODE_STRING, (parser) -> {
+        try {
+            setResiliencyMode(ResiliencyMode.fromName(parser.text()));
+            return null;
+        } catch (IOException e) {
+            throw new IllegalArgumentException("parsing error encountered for the field " + RESILIENCY_MODE_STRING);
+        }
+    }, RESOURCE_LIMITS_STRING, (parser) -> {
+        try {
+            String fieldName = "";
+            XContentParser.Token token;
+            final Map<ResourceType, Double> resourceLimits = new HashMap<>();
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    fieldName = parser.currentName();
+                } else {
+                    resourceLimits.put(ResourceType.fromName(fieldName), parser.doubleValue());
+                }
+            }
+            setResourceLimits(resourceLimits);
+            return null;
+        } catch (IOException e) {
+            throw new IllegalArgumentException("parsing error encountered for the object " + RESOURCE_LIMITS_STRING);
+        }
+    });
```

Comment:
We don't need the `acceptedFieldNames` since these values are present in the fieldParser map keys.

Regarding `fromXContentMap` I think better way to write this would be to write something like following

```
interface FieldParser{
     <T> T parseField(XContentParser parser) throws IOException;
}

static class FieldParserFactory {
    static Optional<FieldParser> fieldParserFor(String fieldName) {
           if (fieldName == RESOURCE_LIMITS) {
               return Optional.of(ResourceLimitsParser);
           } else if (fieldName == RESILIENCY_MODE) {
               return Optional.of(ResiliencyModeParser);
           }
           return Optional.empty();
    }
}

static class ResiliencyModeParser implements FieldParser<ResiliencyMode> {
       ResiliencyMode parseField(XContentParser parser) throws IOException {
            setResiliencyMode(ResiliencyMode.fromName(parser.text()));
       }
}


static class ResourceLimitsParser implements FieldParser<Map<ResourceType, Double>> {
    Map<ResourceType, Double> parseField(XContentParser parser) throws IOException {
    // Implementation goes here
    }
}

```